core/src/main/java/org/apache/cxf/common/util/ClassHelper.java

```diff
@@ -69,19 +76,44 @@ protected Object getRealObjectInternal(Object o) {
     }
 
     public static Class<?> getRealClassFromClass(Class<?> cls) {
```

Comment:
I would suggest to add the overloaded method with accepts `Bus bus`, there are many places where it could be provided from context. What do you think?